compiler/codegen/expr.cpp

```diff
@@ -3494,6 +3494,25 @@ GenRet CallExpr::codegen() {
           ret.val = converted;
         }
       }
+
+      // Handle setting LLVM invariant on const records after
+      // they are initialized
+      if (this->isNamedAstr(astrInit)) {
+        if (isUserDefinedRecord(get(1)->typeInfo())) {
```

Comment:
Why user defined records only?

Reply:
Ah nevermind, I see that this is just a poorly named function now with a TODO to change it

Reply:
isUserDefinedRecord doesn't actually mean user defined records - it started out that way but it actually now means "Records for which we manage creation and destruction in the new way". E.g. range and tuples are "user defined records". There is a TODO comment that it should be renamed "isMemoryManagedRecord" or something along those lines.